### PR TITLE
Simplify EmailLister class

### PR DESCRIPTION
**Pull request by davidglezz**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This unused class can be improved.
| Type?             | refactor
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | unit test
| Possible impacts? | no.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27365)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]
nice refactor 👍

**Review by sowbiba — DISMISSED** [review]

**Review by sowbiba — DISMISSED** [review]
Just asking for tests. Otherwise, very nice work ;)

**Comment by davidglezz** [comment]
@sowbiba Test done

**Review by atomiix — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
@davidglezz sorry for late review

@PrestaShop/prestashop-maintainers agree to merge without QA? this is a refactoring

File: tests/Unit/Core/Email/EmailListerTest.php
Comment (Progi1984) [inline tests/Unit/Core/Email/EmailListerTest.php]:
```suggestion
 */

declare(strict_types=1);

namespace Tests\Unit\Core\Email;
```

**Review by Progi1984 — COMMENTED** [review]
@davidglezz A small comment (and I approve just after)

File: tests/Unit/Core/Email/EmailListerTest.php
Comment (davidglezz) [reply tests/Unit/Core/Email/EmailListerTest.php]: ✅ thanks!

**Review by Progi1984 — APPROVED** [review]

File: src/Core/Email/EmailLister.php
Comment (saulaski) [inline src/Core/Email/EmailLister.php:55]: Exceptions don't have to extend `\Exception` class, but they have to implement `\Throwable` interface at least.
```suggestion
        } catch (Throwable $e) {
```

**Review by saulaski — APPROVED** [review]
Simple and elegant, I like a lot :)

**Comment by saulaski** [comment]
Please try to force push again to restart CI tests

File: src/Core/Email/EmailLister.php
Comment (davidglezz) [reply src/Core/Email/EmailLister.php:55]: Thanks for the review and suggestion, but in this case, `Exception` is `'PrestaShop\PrestaShop\Core\Foundation\Filesystem\Exception'` and not `\Exception`.

My intention is to catch only the known exceptions that `FileSystem::listEntriesRecursively()` throws.

Comment (davidglezz) [reply src/Core/Email/EmailLister.php:55]: I see in PHPStorm

image

I think this warning is caused by the fact that the function is not properly documented, `@throws` is missing in the PHPDoc section of `FileSystem::listEntriesRecursively()`.

**Comment by davidglezz** [comment]
@saulaski I see CI ✅ , am I missing something?

**Review by atomiix — APPROVED** [review]

**Comment by saulaski** [comment]
> @saulaski I see CI white_check_mark , am I missing something?

My bad, CI tests looked stalled when I reviewed. Forget my comment please :)

File: src/Core/Email/EmailLister.php
Comment (saulaski) [reply src/Core/Email/EmailLister.php:55]: No need of phpdoc. Static analysis in phpStorm is supposed to detect potential exceptions thrown by called method.
Perhaps code indexation wasn't completed when you took the screenshot ?
Here the warning from my own IDE:
Capture d’écran de 2022-03-03 15-17-24

**Review by florine2623 — DISMISSED** [review]
Hello @davidglezz ,

Could you create an [issue](https://github.com/PrestaShop/PrestaShop/issues/new/choose) related to this PR please ?
It would be helpful to have steps to reproduce your issue !
Thanks!

**Comment by davidglezz** [comment]
@florine2623 There is no issue, there is nothing to reproduce, I just saw a piece of code that could be improved, and this is my contribution:
- refactoring it
- adding unit tests

I've updated the description table

**Review by florine2623 — COMMENTED** [review]
Hello @davidglezz ,

We need to follow the QA processes.
We need an issue, whether there is something to reproduce or not. You could create a Feature Request to improve the piece of code.
QA and dev will approve or not the issue. Only after that will I be testing your PR :)

Thanks !

**Comment by davidglezz** [comment]
> Refactoring is a disciplined technique for restructuring an existing body of code, altering its internal structure without changing its external behavior.

@florine2623  If it is a refactor, why is required to open an issue if there is no feature request or bug in which to indicate some steps to reproduce?

**Comment by kpodemski** [comment]
Hey @matks 

If I recall correctly refacto, especially with unit tests, doesn't require any Issue?

**Comment by matks** [comment]
> Hey @matks
> 
> If I recall correctly refacto, especially with unit tests, doesn't require any Issue?

Indeed I think a misunderstanding have happened here. Maybe the root of the misunderstanding is that a GitHub issue is a requirement for QA to test a PR but this PR has no such issue and should have not been sent to QA?

**Comment by davidglezz** [comment]
so no QA right?

File: src/Core/Email/EmailLister.php
Comment (MeKeyCool) [inline src/Core/Email/EmailLister.php:54]: IMO "mail_directory" isn't a good name for a file list ^^'

File: src/Core/Email/EmailLister.php
Comment (MeKeyCool) [inline src/Core/Email/EmailLister.php:89]: :+1:

**Review by MeKeyCool — APPROVED** [review]

**Comment by kpodemski** [comment]
@matks 

Can we agree on the next steps regarding this PR? The Core does not use this class. There is no way for QA to test it, and David handled the automated testing part.

Also, I really think that sometimes we could do the "Create an Issue" part ourselves, most of the time is just a minute, and we don't lose time sending pull requests back and forth.

**Comment by davidglezz** [comment]
As it is not used, the best action could be delete the file/class, thus we contribute to the cleaning.

No problem, feel free to close this PR =)

**Pull request closed without merging**
